meroxa/turbine-py
# feat: Turbine-core implementation for Turbine-PY

# Description

As part of Turbine core project, we are making following changes to the turbine-py library: 

- [x] Update the public library functions to map to the associated gRPC calls. - https://github.com/meroxa/turbine-py/issues/208
- [x] [Expose record function from runner - https://github.com/meroxa/turbine-py/issues/209
- [x]  Implement GRPC client init - https://github.com/meroxa/turbine-py/issues/207 

The implementation plan - 

https://www.notion.so/Turbine-Core-Migration-Implementation-Plan-5d2e805db7bc47b3b53513b3442f12e3

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

To test the following code - 

1.  Modify setup.config to point to pkg/ code (turbine-core implementation) instead of src/ code (old code) 
2. Run - `pip install -U turbine-py/ ` on your turbine-py
3. You should be able to now execute new core commands, such as `run` 
4. You can grab latests cli changes for turbine-py that implement run command for core - https://github.com/meroxa/cli/pull/715
5. Create a data app and run "meroxa app run ."
